internal/handlers/pg/fetch.go

```diff
@@ -31,27 +30,15 @@ type sqlParam struct {
 }
 
 // fetch fetches all documents from the given database and collection.
-// If collection doesn't exist it returns an empty slice and no error.
 //
-// TODO https://github.com/FerretDB/FerretDB/issues/372
-func (h *Handler) fetch(ctx context.Context, param sqlParam) ([]*types.Document, error) {
-	// Special case: check if collection exists at all
-	collectionExists, err := h.pgPool.CollectionExists(ctx, param.db, param.collection)
-	if err != nil {
-		return nil, lazyerrors.Error(err)
-	}
-	if !collectionExists {
-		h.l.Info(
-			"Collection doesn't exist, handling a case to deal with a non-existing collection.",
-			zap.String("schema", param.db), zap.String("table", param.collection),
-		)
-		return []*types.Document{}, nil
-	}
-
-	res, err := h.pgPool.QueryDocuments(ctx, param.db, param.collection, param.comment)
-	if err != nil {
-		return nil, lazyerrors.Error(err)
-	}
-
-	return res, nil
+// If an error occurs before the fetching, the error is returned immediately.
+// The returned channel is always non-nil.
+// The channel is closed when all documents are sent; the caller should always drain the channel.
+// If an error occurs during fetching, the last message before closing the channel contains an error.
+// Context cancelation is not considered an error.
+//
+// If the collection doesn't exist, fetch returns a closed channel and no error.
+//
+func (h *Handler) fetch(ctx context.Context, querier pgxtype.Querier, param sqlParam) (<-chan pgdb.FetchedDocs, error) {
```

Comment:
Do we need `fetch` anymore if all that it does now is call `QueryDocuments`?

Reply:
Fixed